source/best-practices/software-installation.rst

```diff
@@ -0,0 +1,94 @@
+软件安装最佳实践
+================
+
+:本节贡献者: `姚家园 <https://github.com/core-man>`__ （作者）、`田冬冬 <https://me.seisman.info>`__ （作者）
+:最近更新日期: 2021-01-12
+:预计阅读时间: X 分钟
+
+软件安装方式
+------------
+
+Linux 或 macOS 系统下安装软件通常有如下几种方式：
+
+1.  使用系统自带的应用商店（App Store）在图形界面下安装软件
+2.  使用软件包管理器通过命令行安装软件。例如，Ubuntu/Debian 的 ``apt``\ 、
+    Fedora 的 ``dnf``\ 、macOS 的 ``brew``\ 。这些软件包管理器既可以安装
+    官方源提供的软件，也可以安装第三方源提供的软件。最好选择可信赖
+    的第三方源。
+3.  直接下载闭源软件或商业软件提供的二进制包进行安装，如 Linux 下的 ``deb`` 和 ``rpm``\ 、
+    macOS 下的 ``dmg``\ 。
+4.  某个编程语言的包，通常可以使用该编程语言自带的包管理工具进行安装，比如
+    Python 语言的包管理器 ``conda`` 和 ``pip``
+5.  某些软件提供了二进制的压缩包，则解压后直接放在某个目录，然后修改环境变量
+    **PATH** 即可使用
+6.  下载软件源码、编译并安装
+
+绝大多数非科研类开源软件都可以通过方式 1—3 安装；多数商业软件
+都可以通过方式 1—3 和方式 5 安装；而对于地震学科研软件而言，
+则可以通过方式 4（很多地震学软件是用 Python 语言写的，因而可以通过 ``conda``
+或 ``pip`` 安装）、方式 5 或方式 6 进行安装。
+
+方式 1—4 会将软件安装到默认的位置，通常用户无法控制也无需关心其具体安装位置。
+而对于方式 5 和 6，用户可选择将软件安装到任意位置。为了便于安装与管理科研
+相关软件，我们总结了如下最佳实践。
+
+软件安装路径
+------------
+
+软件安装相关的路径如下：
+
+- :file:`~/software/`
+- :file:`~/opt/`
+- :file:`~/src/`
+- :file:`~/bin/`
+- :file:`/opt/`
+
+软件包
+^^^^^^^
+
+软件的原始压缩包可以保存在 :file:`~/software/` 的子目录下, 相当于做一个备份，
+如 :file:`~/software/TauP-2.4.5.tgz`\ 。
+
+解压即用的软件
+^^^^^^^^^^^^^^
+
+某些软件提供了二进制包，可以解压即用，可安装在 :file:`~/opt/` 的子目录下，
+如 :file:`~/opt/TauP`\ 。
+
+从源码编译的软件
+^^^^^^^^^^^^^^^^
+
+需要从源码编译的软件，可将源码目录放在 :file:`~/src/` 的子目录下，如 :file:`~/src/fk`\ 、
+:file:`~/src/GMT`\ 、:file:`~/src/distaz`\ 。
+
+若该软件不需安装，则编译链接后生成的可执行文件在该软件目录下或其子目录 :file:`bin` 下，
+如 :file:`~/src/fk`\ 、:file:`~/src/CPS/bin`\ 。如果该软件只是一个简单的小工具，
+比如单个源码文件或单个可执行文件，则可以将生成的可执行文件复制至 :file:`~/bin/` 目录下，
+如 :file:`~/bin/distaz`\ 。
```

Comment:
How do we deal with fk? its binary files are in the source dir.

Reply:
Considering that fk provides several commands and scripts, perhaps the easiest way is still to add the fk directory to PATH

Reply:
Shall we make a bin folder under ``fk` instead? although this is not related to the topic here.

Reply:
Maybe no.